golem-ts-agentic/packages/golem-ts-sdk/src/internal/schema.ts

```diff
@@ -474,18 +464,51 @@ export function buildOutputSchema(
 }
 
 function getMultimodalDetails(
-  types: Type.Type[],
+  type: Type.Type
 ): Either.Either<[string, ElementSchema, TypeInfoInternal][], string> {
+
+  const multimodalTypes =
+    type.kind === 'union' ? getTaggedUnion(type.unionTypes) : getTaggedUnion([type]);
+
+  if (Either.isLeft(multimodalTypes)) {
+    return Either.left(
+      `failed to generate the multimodal schema: ${multimodalTypes.val}`,
+    );
+  }
+
+  const taggedUnionOpt = multimodalTypes.val;
+
+  if (Option.isNone(taggedUnionOpt)) {
+    return Either.left(
+      `multimodal type is not a tagged union: ${multimodalTypes.val}`,
+    );
```

Comment:
This indicates its not a tagged union.. There is no `tag` associated with the types.